lib/brook/storage/ets.ex

```diff
@@ -10,64 +10,64 @@ defmodule Brook.Storage.Ets do
   """
   use GenServer
   require Logger
+  import Brook.Config, only: [registry: 1]
   @behaviour Brook.Storage
 
-  @table :brook_test_value
-
   @impl Brook.Storage
-  def persist(%Brook.Event{} = event, collection, key, value) do
-    GenServer.call(__MODULE__, {:persist, event, collection, key, value})
+  def persist(instance, %Brook.Event{} = event, collection, key, value) do
+    GenServer.call(via(registry(instance)), {:persist, event, collection, key, value})
   end
 
   @impl Brook.Storage
-  def delete(collection, key) do
-    GenServer.call(__MODULE__, {:delete, collection, key})
+  def delete(instance, collection, key) do
+    GenServer.call(via(registry(instance)), {:delete, collection, key})
   end
 
   @impl Brook.Storage
-  def get(collection, key) do
-    GenServer.call(__MODULE__, {:get, collection, key})
+  def get(instance, collection, key) do
+    GenServer.call(via(registry(instance)), {:get, collection, key})
   end
 
   @impl Brook.Storage
-  def get_events(collection, key) do
-    GenServer.call(__MODULE__, {:get_events, collection, key})
+  def get_events(instance, collection, key) do
+    GenServer.call(via(registry(instance)), {:get_events, collection, key})
   end
 
   @impl Brook.Storage
-  def get_all(collection) do
-    GenServer.call(__MODULE__, {:get_all, collection})
+  def get_all(instance, collection) do
+    GenServer.call(via(registry(instance)), {:get_all, collection})
   end
 
   @impl Brook.Storage
-  def start_link(_opts) do
-    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  def start_link(args) do
+    instance = Keyword.fetch!(args, :instance)
+    GenServer.start_link(__MODULE__, args, name: via(registry(instance)))
   end
 
   @impl GenServer
-  def init([]) do
-    :ets.new(@table, [:named_table, :set, :protected])
+  def init(_args) do
+    table = :ets.new(nil, [:set, :protected])
```

Comment:
Will it make it difficult to read from this table without giving it some form of known name or are you going to expose that from the process state somehow (or just don’t care to make it readable outside the owning process)?

Reply:
Don't care at the moment, but if you have a reason to care we can give it a name.